plugins/repository-hdfs/build.gradle

```diff
@@ -205,10 +258,72 @@ if (secureFixtureSupported) {
+  integTestSecureRunner.exclude('**/Ha*TestSuiteIT.class') // Ignore HA integration Tests
 } else {
   integTestSecure.enabled = false
 }
 
+// Create a Integration Test suite just for HA related security based tests
+// The following closure must execute before the afterEvaluate block from integTestSecureHA:
+project.afterEvaluate {
+  Path elasticsearchKT = project(':test:fixtures:krb5kdc-fixture').buildDir.toPath().resolve("keytabs").resolve("elasticsearch.keytab").toAbsolutePath()
+  Path krb5conf = project(':test:fixtures:krb5kdc-fixture').buildDir.toPath().resolve("conf").resolve("krb5.conf").toAbsolutePath()
+
+  project.integTestSecureHaCluster.dependsOn(project.bundlePlugin)
+  project.integTestSecureHa.clusterConfig.plugin(project.path)
+  project.integTestSecureHa.clusterConfig.extraConfigFile("repository-hdfs/krb5.keytab", "${elasticsearchKT}")
+
+  final String baseClusterJvmArgs = "-Xms" + System.getProperty('tests.heap.size', '512m') +
+          " " + "-Xmx" + System.getProperty('tests.heap.size', '512m') +
+          " " + "-Djava.security.krb5.conf=${krb5conf}" +
+          " " + System.getProperty('tests.jvm.argline', '')
+  final String jvmArgs
+  if (project.rootProject.ext.javaVersion == JavaVersion.VERSION_1_9) {
+    jvmArgs = baseClusterJvmArgs + " " + '--add-opens=java.security.jgss/sun.security.krb5=ALL-UNNAMED'
+  } else {
+    jvmArgs = baseClusterJvmArgs
+  }
+
+  project.integTestSecureHa.clusterConfig.jvmArgs = jvmArgs
+
+  // also set the kerberos settings for the integration test JVM since we'll need to auth to HDFS to trigger namenode failovers.
+  project.integTestSecureHaRunner.systemProperty "test.krb5.principal.es", "elasticsearch@${realm}"
+  project.integTestSecureHaRunner.systemProperty "test.krb5.principal.hdfs", "hdfs/hdfs.build.elastic.co@${realm}"
+  project.integTestSecureHaRunner.jvmArg "-Djava.security.krb5.conf=${krb5conf}"
+  if (project.rootProject.ext.javaVersion == JavaVersion.VERSION_1_9) {
+    project.integTestSecureHaRunner.jvmArg '--add-opens=java.security.jgss/sun.security.krb5=ALL-UNNAMED'
+  }
+  // We want to use the temporary directory to access the keytab file for failing over HDFS remotely.
+  // To do this, we need to ignore the non-empty work dir for the test, and trust this build to wipe it before hand.
+  project.integTestSecureHaRunner.onNonEmptyWorkDirectory 'ignore'
+}
+
+RestIntegTestTask integTestSecureHa = project.tasks.create('integTestSecureHa', RestIntegTestTask.class) {
+  description = "Runs rest tests against an elasticsearch cluster with HDFS configured with HA Namenode and secured by MIT Kerberos."
+}
+
+// set the keytab file somewhere we can access it from the test. In this case, place it into the temporary directory:
+project.integTestSecureHaRunner.doFirst {
+  Path tempDir = project.buildDir.toPath().resolve("testrun").resolve("integTestSecureHaRunner").resolve("J0").resolve("temp")
```

Comment:
ah...this is so hacked.... how about adding the folder with the keytab file to the test classpath instead?